cluster/charts/crossplane/values.yaml.tmpl

```diff
@@ -9,6 +9,9 @@ image:
 
 args: {}
 
+provider:
+  packages: []
```

Comment:
nitpick: what would you think about switching these (i.e. `packages.provider`)? Then we could also support installing other package types in the future by just adding another field to this stanza

Reply:
I went back and forth about this and am still on the fence. At one side, as you point out a `Provider` is a package, so it makes to have `packages.providers` and/or `packages.configurations`. On the other hand, package is what goes to `spec.package` of those resources and users don't really know about the concept that `Provider`s and `Configuration`s are package types, `package` for them means `spec.package` which is actually `image`. So, `provider.packages` makes sense if you treat `package` as the image because the values are in fact image paths and also `provider.packages` read like what we want it to be.

Reply:
Hmm thinking of the CLI semantics though we say `kubectl crossplane install <package-type> <package-image>`. So that would be (package) --> (package type) --> (package image name), which I feel like matches `packages.providers` and `packages.configurations`.

Reply:
I slightly lean toward the current `provider.packages` here, for the reasons @muvaf points out. This seems a little more like a UX of "installing a provider, which is in package foo", vs "installing a package, which is provider foo". The former meshes more with our desired UX of the user "installing a provider", not "installing a package".

Reply:
What would you think of just `providers`? I don't think we will have other fields under `provider` here, which is why is made more sense to me to have `packages.___` as we could have multiple fields under than (i.e. `providers`, `configurations`, etc.). Not a huge deal so fine to defer if y'all like it as is :+1:

Reply:
I'd be fine with just `providers` too. I'll leave this for @muvaf to decide before he merges.

Reply:
There is also the option of having:
```
providers:
  - package: crossplane/provider-gcp:master
  - package: crossplane/provider-aws:master
```

which paves the way for further customization. But I don't like the way users have to pass in the values:
```
--set providers[0].package='crossplane/provider-gcp:master',providers[1].package='crossplane/provider-aws:master'
```

Tried a few things but it seems that's the support Helm has for passing in an array of objects. I'll merge this PR but let me know if you find a nice UX for that structure, we can revisit.